scene/gui/flow_container.h

```diff
@@ -0,0 +1,87 @@
+/*************************************************************************/
+/*  flow_container.h                                                     */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef FLOW_CONTAINER_H
+#define FLOW_CONTAINER_H
+
+#include "scene/gui/container.h"
```

Comment:
This can be moved to .cpp file. Just declare an empty `Container` class in the header.

Reply:
I just tried this, but I couldn't get it to compile. 
I think we should leave it like it is now, as for other containers the inclusion of `"scene/gui/container.h"` also happens in the header file (for example in box_container.h)

Reply:
It does compile for me. When you move the include, you need to add `class Container;` to your header, so that it knows about its existence.

If you move the include, FlowContainer doesn't need to be recompiled when Container is modified. We have lots of unnecessary includes in header files, which make compile times longer for no reason. BoxContainer is actually an example of bad include 🙃

Reply:
While working on some other PR I noticed that this could cause problems since the forward declaration of a class which is used as a base class shouldn't work actually (have no idea why it does though). When I do the same thing for MarginContainer, which is quite deeply integrated into the editor (used at many places) the compilation fails. It looks like that it fails when another class is inheriting from the class which itself is inheriting from a forward-declared class, so I propose we remove it.

Reply:
Yeah I agree, the base class shouldn't be forward declared.